src/SmartNumber.php

```diff
@@ -153,7 +183,12 @@ public function getFraction()
 	 */
 	public function isInteger(): bool
 	{
-		return $this->integer !== null && ($this->input === (string) $this->integer || $this->getFraction()[1] === '1');
+		try {
+			$this->decimal->toScale(0);
+			return true;
+		} catch (RoundingNecessaryException $e) {
+			return false;
+		}
```

Comment:
I think this is safer for future compatibility:

```suggestion
		} catch (RoundingNecessaryException $e) {
		}
		return false;
```

Reply:
Done.